src/arithmetic/arithmetic_expression_construct.c

```diff
@@ -397,9 +397,24 @@ static AR_ExpNode *_AR_ExpFromMapProjection(const cypher_astnode_t *expr) {
 	const cypher_astnode_t *selector = NULL;
 	unsigned int n_selectors = cypher_ast_map_projection_nselectors(expr);
 
-	AR_ExpNode *op = AR_EXP_NewOpNode("tomap", true, n_selectors * 2);
-	AR_ExpNode **children = op->op.children;
+	AR_ExpNode *propertiesOp = NULL;
+	AR_ExpNode *tomapOp = NULL;
 
+	// Count the number of selectors of type CYPHER_AST_MAP_PROJECTION_ALL_PROPERTIES, because these are not children of tomap OpNode
+	uint allProps_selectors = 0;
+	for(uint i = 0; i < n_selectors; i++) {
+		selector = cypher_ast_map_projection_get_selector(expr, i);
+		t = cypher_astnode_type(selector);
+		if(t == CYPHER_AST_MAP_PROJECTION_ALL_PROPERTIES) {
+			// { .* }
+			allProps_selectors++;
+		}
+	}
+
+	tomapOp = AR_EXP_NewOpNode("tomap", true, (n_selectors - allProps_selectors) * 2);
```

Comment:
What if `n_selectors - allProps_selectors == 0` see additional comments below

Reply:
You are right, that was not considered. Now, tomapOp is initialized only if ``` (n_selectors - allProps_selectors) > 0```.
[8814e48](https://github.com/RedisGraph/RedisGraph/pull/2698/commits/8814e4827683d32aca947ccbe1fac3817267a3d8)

Reply:
I removed ```(n_selectors - allProps_selectors) > 0``` and there is no problem when it is zero.
tomap() function accepts 0 arguments, tomapOp is initialized always and the test don't report errors.